Fix bad updates by Dependabot.

When Dependabot updates references in a C# project that is referenced in another project, then it only updates lock file for 1st project and not the dependent project leading to this error when running tests IN workflows: `error NU1004: The project references X.Y whose dependencies has changed.`.

To fix this, we could always use `--force-evaluate` when restoring dependencies but this masks the problem IMHO.

I think it's better to commit any modified lock files and push to PR branch. This is useful to do also when a linter modifies files.

--- a/.github/workflows/codeql-analysis.yml
+++ b/.github/workflows/codeql-analysis.yml
@@ -106,6 +106,14 @@ jobs:
         run: |
           if [[ "${{ github.event.pull_request.user.login }}" == 'dependabot[bot]' ]] || [[ "${{ github.actor }}" == 'dependabot[bot]' ]]; then
             dotnet restore --force-evaluate
+            if git status --porcelain | grep -q .; then
+              git config --local user.email "${{ github.actor }}@users.noreply.github.com"
+              git add ./\*.lock.json
+              git commit -m "Committing changes to lock-files" -S
+              git push
+            else
+              echo "No changes detected"
+            fi
           else
             dotnet restore --locked-mode
           fi

--- a/.github/workflows/codeql-analysis.yml
+++ b/.github/workflows/codeql-analysis.yml
@@ -107,7 +107,8 @@ jobs:
           if [[ "${{ github.event.pull_request.user.login }}" == 'dependabot[bot]' ]] || [[ "${{ github.actor }}" == 'dependabot[bot]' ]]; then
             dotnet restore --force-evaluate
             if git status --porcelain | grep -q .; then
-              git config --local user.email "${{ github.actor }}@users.noreply.github.com"
+              git config --global user.name "${{ github.actor }}"
+              git config --global user.email "${{ github.actor }}@users.noreply.github.com"
               git add ./\*.lock.json
               git commit -m "Committing changes to lock-files" -S
               git push

--- a/.github/workflows/codeql-analysis.yml
+++ b/.github/workflows/codeql-analysis.yml
@@ -109,9 +109,9 @@ jobs:
             if git status --porcelain | grep -q .; then
               git config --global user.name "${{ github.actor }}"
               git config --global user.email "${{ github.actor }}@users.noreply.github.com"
-              git config --local commit.gpgsign false
+              git config --global commit.gpgsign false
               git add ./\*.lock.json
-              git commit -m "Committing changes to lock-files" -S
+              git commit -m "Committing changes to lock-files"
               git push
             else
               echo "No changes detected"

--- a/.github/workflows/codeql-analysis.yml
+++ b/.github/workflows/codeql-analysis.yml
@@ -112,7 +112,7 @@ jobs:
               git config --global commit.gpgsign false
               git add ./\*.lock.json
               git commit -m "Committing changes to lock-files"
-              git push
+              git push origin HEAD:${{ github.event.pull_request.head.ref }}
             else
               echo "No changes detected"
             fi

--- a/.github/workflows/codeql-analysis.yml
+++ b/.github/workflows/codeql-analysis.yml
@@ -109,6 +109,7 @@ jobs:
             if git status --porcelain | grep -q .; then
               git config --global user.name "${{ github.actor }}"
               git config --global user.email "${{ github.actor }}@users.noreply.github.com"
+              git config --local commit.gpgsign false
               git add ./\*.lock.json
               git commit -m "Committing changes to lock-files" -S
               git push

--- a/.github/workflows/codeql-analysis.yml
+++ b/.github/workflows/codeql-analysis.yml
@@ -19,7 +19,7 @@ concurrency:
   cancel-in-progress: true
 
 permissions:
-  contents: read
+  contents: write
 
 jobs:
   analyze:

--- a/.github/workflows/infersharp.yml
+++ b/.github/workflows/infersharp.yml
@@ -3,7 +3,7 @@ name: "Infersharp"
 on: [workflow_dispatch, pull_request ]
 
 permissions:
-  contents: read
+  contents: write
   security-events: write
 
 jobs:

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -81,6 +81,14 @@ jobs:
         if: env.ENABLED == '1' && steps.pr-check.outputs.pr_found == 'true'
         run: |
             dotnet restore --force-evaluate
+            if git status --porcelain | grep -q .; then
+              git config --local user.email "${{ github.actor }}@users.noreply.github.com"
+              git add ./\*.lock.json
+              git commit -m "Committing changes to lock-files" -S
+              git push
+            else
+              echo "No changes detected"
+            fi
 
       - name: Install dependencies
         if: env.ENABLED == '1' && steps.pr-check.outputs.pr_found == 'false'

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -87,7 +87,7 @@ jobs:
               git config --global commit.gpgsign false
               git add ./\*.lock.json
               git commit -m "Committing changes to lock-files"
-              git push
+              git push origin HEAD:${{ github.event.pull_request.head.ref }}
             else
               echo "No changes detected"
             fi

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -84,6 +84,7 @@ jobs:
             if git status --porcelain | grep -q .; then
               git config --global user.name "${{ github.actor }}"
               git config --global user.email "${{ github.actor }}@users.noreply.github.com"
+              git config --local commit.gpgsign false
               git add ./\*.lock.json
               git commit -m "Committing changes to lock-files" -S
               git push

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -8,7 +8,7 @@ on:
   push:
   pull_request:
 permissions:
-  contents: read
+  contents: write
   pull-requests: write
 
   # https://www.meziantou.net/how-to-cancel-github-workflows-when-pushing-new-commits-on-a-branch.htm

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -82,7 +82,8 @@ jobs:
         run: |
             dotnet restore --force-evaluate
             if git status --porcelain | grep -q .; then
-              git config --local user.email "${{ github.actor }}@users.noreply.github.com"
+              git config --global user.name "${{ github.actor }}"
+              git config --global user.email "${{ github.actor }}@users.noreply.github.com"
               git add ./\*.lock.json
               git commit -m "Committing changes to lock-files" -S
               git push

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -84,9 +84,9 @@ jobs:
             if git status --porcelain | grep -q .; then
               git config --global user.name "${{ github.actor }}"
               git config --global user.email "${{ github.actor }}@users.noreply.github.com"
-              git config --local commit.gpgsign false
+              git config --global commit.gpgsign false
               git add ./\*.lock.json
-              git commit -m "Committing changes to lock-files" -S
+              git commit -m "Committing changes to lock-files"
               git push
             else
               echo "No changes detected"

